execution/executor-types/src/lib.rs

```diff
@@ -296,35 +296,35 @@ pub struct ChunkCommitNotification {
 #[derive(Debug, Default, PartialEq, Eq, Clone, Serialize, Deserialize)]
 pub struct StateComputeResult {
     /// transaction accumulator root hash is identified as `state_id` in Consensus.
-    root_hash: HashValue,
+    pub root_hash: HashValue,
```

Comment:
why do we need to add pub here?

Reply:
These are made pub so i can do:
```rust
    let state_compute_result = StateComputeResult {
        compute_status_for_input_txns: vec![TransactionStatus::Keep(ExecutionStatus::Success); 3],
        events: vec![event_0.clone(), event_1.clone(), event_2.clone()],
        ..StateComputeResult::dummy()
    };
```
in the tests.
Otherwise I would have to make a few functions:
- `StateComputeResult::dummy_with_compute_status()`
- `StateComputeResult::dummy_with_events()`
- `StateComputeResult::dummy_with_compute_status_and_events()`
which is awkward.

Reply:
If this is only used for test, I'd prefer to add few more functions..
You probably can only add StateComputeResult::dummy_with_compute_status_and_events() since you can pass vec![] as one of its parameters.

Reply:
also not needed any more since we are moving the filter to `assemble_ledger_diff`...